src/main/java/org/wso2/carbon/module/swiftiso20022/MT940FormatValidator.java

```diff
@@ -0,0 +1,134 @@
+        org.apache.axis2.context.MessageContext axis2MessageContext = ((Axis2MessageContext) messageContext)
+                .getAxis2MessageContext();
+        Optional<String> payload = ConnectorUtils.buildMessagePayloadFromMessageContext(axis2MessageContext);
+        if (payload.isPresent()) {
+            String[] lines = payload.get().split("\n");
+            ErrorModel validationResponse = validateMT940(messageContext, lines);
+
+            if (validationResponse.isError()) {
+                this.log.error(validationResponse.getErrorMessage());
+                ConnectorUtils.appendErrorToMessageContext(messageContext, validationResponse.getErrorCode(),
+                        validationResponse.getErrorMessage());
+                this.handleException(ConnectorConstants.ERROR_VALIDATION_FAILED, messageContext);
+            }
+
+        } else {
+            this.log.error(ConnectorConstants.ERROR_MISSING_PAYLOAD);
+            ConnectorUtils.appendErrorToMessageContext(messageContext, ConnectorConstants.MISSING_REQUEST_PAYLOAD,
+                    ConnectorConstants.ERROR_MISSING_PAYLOAD);
+            this.handleException(ConnectorConstants.ERROR_MISSING_PAYLOAD, messageContext);
+        }
+    }
+
+    /**
+     * Method to validate the MT940 format.
+     *
+     * @param messageContext  Message context
+     * @param lines           Payload lines
+     * @return                Error model
+     */
+    private ErrorModel validateMT940(MessageContext messageContext, String[] lines) {
+        ErrorModel errorModel = new ErrorModel();
+        Map<String, Object> extractFields = extractFields(lines);
+
+        if (!MT940ValidationUtils.validateC1Rule(lines)) {
+            return new ErrorModel(ConnectorConstants.ERROR_C24,
+                    ConnectorConstants.ERROR_FIELD_86);
+        }
+
+        if (!MT940ValidationUtils.validateC2Rule(lines)) {
+            return new ErrorModel(ConnectorConstants.ERROR_C277,
+                    ConnectorConstants.ERROR_BALANCES);
+        }
+
+        errorModel = MT940ValidationUtils.validateMT940Format(extractFields);
+        if (errorModel.isError()) {
+            this.log.error(errorModel.getErrorMessage());
+            ConnectorUtils.appendErrorToMessageContext(messageContext, errorModel.getErrorCode(),
+                    errorModel.getErrorMessage());
+            this.handleException(errorModel.getErrorMessage(), messageContext);
+        }
+
+        errorModel.setIsError(false);
+        return errorModel;
+    }
+
+    /**
+     * Method to extract the fields from the payload.
+     *
+     * @param lines  Payload lines
+     * @return       Extracted fields
+     */
+    private Map<String, Object> extractFields(String[] lines) {
+        Map<String, Object> fields = new HashMap<>();
+        List<String> statementLine = new ArrayList<>();
+
+        for (String line : lines) {
+            if (line.startsWith(ConnectorConstants.MT940_TRANSACTION_REF)) {
```

Comment:
Better to use a switch case statement

Reply:
Fixed in https://github.com/wso2-extensions/mediation-swift-iso20022-module/pull/2/commits/18ec29ed3926964e6b999c6fd5d385b24c1f8fa5